### PR TITLE
Add strong typing to common event properties

### PR DESCRIPTION
**Pull request by guperrot**
Test app U.I. in a separate PR.

### CODE REVIEW

**Comment by guperrot** [comment]
#833 should be approved before merging this

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/PropertyConfiguratorTest.java [hunk 1]
```diff
@@ -364,6 +381,50 @@ public void trackEventWithEmptyProperties() {
...
+
+        /* Set common properties for various types. Some of which are invalid. */
+        target.getPropertyConfigurator().setEventProperty("myString", "hello");
+        target.getPropertyConfigurator().setEventProperty("myNullString", (String) null);
+        target.getPropertyConfigurator().setEventProperty("myTrue", true);
+        target.getPropertyConfigurator().setEventProperty("myFalse", false);
+        target.getPropertyConfigurator().setEventProperty("myLong", 123);
+        target.getPropertyConfigurator().setEventProperty("myDate", new Date(456));
+        target.getPropertyConfigurator().setEventProperty("myNullDate", (Date) null);
+        target.getPropertyConfigurator().setEventProperty("myDouble", -3.14);
```
Comment (dhei) [inline sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/PropertyConfiguratorTest.java]: Add a case for scientific notation of double like "1.2345e" would be great

Comment (guperrot) [reply sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/PropertyConfiguratorTest.java]: Another method has a simple double so will just change this one

File: sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/PropertyConfiguratorTest.java [hunk 2]
```diff
@@ -364,6 +381,50 @@ public void trackEventWithEmptyProperties() {
...
+
+        /* Create transmission target. */
+        AnalyticsTransmissionTarget target = Analytics.getTransmissionTarget("test");
+
+        /* Set common properties for various types. Some of which are invalid. */
+        target.getPropertyConfigurator().setEventProperty("myString", "hello");
+        target.getPropertyConfigurator().setEventProperty("myNullString", (String) null);
+        target.getPropertyConfigurator().setEventProperty("myTrue", true);
+        target.getPropertyConfigurator().setEventProperty("myFalse", false);
+        target.getPropertyConfigurator().setEventProperty("myLong", 123);
```
[the remark this answers is not in the document]
Comment (guperrot) [reply sdk/appcenter-analytics/src/test/java/com/microsoft/appcenter/analytics/PropertyConfiguratorTest.java]: Using 2^64 -1 then ;)

**Pull request merged**